src/NuGet.Core/NuGet.Configuration/PackageSourceMapping/PackageSourceMapping.cs

```diff
@@ -20,25 +22,30 @@ public class PackageSourceMapping
         /// </summary>
         internal IReadOnlyDictionary<string, IReadOnlyList<string>> Patterns { get; }
 
-        private Lazy<SearchTree> SearchTree { get; }
+        private Lazy<SearchTree?> SearchTree { get; }
```

Comment:
Would it be more natural to have the lazy be null altogether? This way you have an extra object even if PSM is not enabled (majority of the scenarios right now).